src/main/java/io/horizen/lambo/car/transaction/CarDeclarationTransaction.java

```diff
@@ -27,10 +27,9 @@
 // As outputs it contains possible RegularBoxes(to pay fee and change) and new CarBox entry.
 // No specific unlockers to parent class logic, but has specific new box.
 // TODO: add specific mempool incompatibility checker to deprecate keeping in the Mempool txs that declare the same Car.
-public final class CarDeclarationTransaction extends AbstractRegularTransaction {
+public final class CarDeclarationTransaction<P extends Proposition, B extends NoncedBox<P>> extends AbstractRegularTransaction {
```

Comment:
I don't think that it's correct to define generics here, because in one of the parent classes the proper values were already set to the templates. The previous error with `instanceof` occurs because of the generics, which are not covariant.
I would suggest to revert current line changes and in the error place use stuff like `CarDeclarationTransaction.class.isInstance(t)`.
Should be fine.

Reply:
done!